Update `assert` schema and associated tests, fix typos

alancai98 opened the pull request:
Resolves #4.

Remodels the `assert` field of the test to be either a struct or list of structs. This provides additional functionality for future tests and tested properties. I included some example test files that have `assert` as a struct and some as a list of structs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

jpschorr reviewed (commented):
#4 says "Remodel test data model assert to be **either a struct or list of structs**"

But this seems to rewrite all tests to be a list of structs even though there is only one assertion.

From the issue, I would expect all the tests to just be a struct as they were before, but:
- for the schema.md to show both the single struct and list of struct options. and
- for the parsing code in the conformance-test-generator to (eventually) be able to handle either the single or list case and for `ParsePassTestCase` to have an member that `assertons: Vec<Assertion>` or similar

alancai98 commented:
Updated the PR to:

- change `assert` to be either a struct or list of structs (rather than just list of structs)
- updated the `partiql-tests-schema-proposal.md` to include examples of `assert` fields with a struct or list of structs
- updated some tests `assert` fields to use a struct
  - I left `literals.ion`'s test `assert` fields as lists so the test runner implementations could ensure they can parse both structs and list of structs

am357 commented on partiql-tests-schema-proposal.md:
```diff
@@ -23,45 +23,82 @@ The following is an abstraction to describe current and future tests we will hav
     statement: <string>,
     <other fields relevant for corresponding test_category>
-    assert: <struct>
+    assert: <struct | <list<struct>>>
```
nit: `<struct> | <list<struct>>`?

alancai98 replied:
Updated in latest commit b6ea14c. Also updated in a couple other places for `env`.
